### DISCUSSION TITLE
Feature request: responsive values

### DISCUSSION THREAD

**Discussion by nandorojo** [tamagui/tamagui#264] [Ideas]
Dripsy offers this API for responsive styles:

```tsx
<View 
  sx={{ bg: ['blue', 'red'] }}
/>
```

I've found this more intuitive and syntactically pretty than Tamagui's. I'd rather specify all the styles for a given style prop in one place. This would also allow more back-and-forth between Tamagui and Dripsy, since they have an overlapping userbase. I believe Stitches also has this approach as an option.

```tsx
<Stack 
  bg={{ $base: 'blue', $md: 'red' }}
/>
```

This API would be nice. @natew and I briefly discussed this on Discord.

**Comment by natew** [comment]
This is definitely possible, just requires a decent amount of typescript foo, and a nicely written loop to turn sx style back into what Tamagui expects (all in a simple functional variant). 

I’d like to do this and tailwind, but focused on a few more 1.0 bugs that I think I can only solve effectively. Until then if anyone starts a branch I’ll lend guidance until I can revisit.

**Reply by nandorojo** [reply]
What files / places would I look to do this?
